pkg/service/log-interceptor.go

```diff
@@ -0,0 +1,51 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+
+	"connectrpc.com/connect"
+)
+
+type logInterceptor struct {
+	log *slog.Logger
+}
+
+func newLogRequestInterceptor(log *slog.Logger) *logInterceptor {
+	return &logInterceptor{
+		log: log,
+	}
+}
+
+func (i *logInterceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
+	return connect.UnaryFunc(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		log := i.log.With("procedure", req.Spec().Procedure)
+
+		if i.log.Enabled(ctx, slog.LevelDebug) {
+			marshaled, _ := json.Marshal(req.Any())
+			log = log.With("body", marshaled)
```

Comment:
not sure if unmarshaling is required, https://github.com/metal-stack/go-ipam/blob/master/cmd/server/server.go#L140